tasks/mrrc-collect-params/mrrc-collect-params.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: mrrc-collect-params
+spec:
+  params:
+    - name: dataJsonPath
+      description: path to data json file
+  results:
+    - name: mrrcParamFilePath
+      description: path of the env file for following tasks
+    - name: cosignPubKeyFile
+      description: the public key file name for cosign verification
+    - name: cosignPubKeySecret
+      description: the secret name for cosign verify public key
+    - name: charonAWSSecret
+      description: the secret name for charon aws credential file
+  steps:
+    - name: mrrc-collect-params
+      image: quay.io/konflux-ci/yq:latest
+      script: |
+        set -eux
+
+        cd $(workspaces.data.path)
+        mrrcEnvFile="./mrrc.env"
+        if [ -f $mrrcEnvFile ]; then
+          rm $mrrcEnvFile
+        fi
+
+        zipRegistry=$(yq -r '.zipRegistry' $(params.dataJsonPath))
+        echo "export MRRC_ZIP_REGISTRY=$zipRegistry" >> $mrrcEnvFile
+
+        environment=$(yq -r '.mrrc.environment' $(params.dataJsonPath))
+        release=$(yq -r '.mrrc.release' $(params.dataJsonPath))
+        target=$environment-maven-$release
+        echo "export MRRC_TARGET=$target" >> $mrrcEnvFile
+
+        productName=$(yq -r '.mrrc.product.name' $(params.dataJsonPath))
+        productVersion=$(yq -r '.mrrc.product.version' $(params.dataJsonPath))
+        echo "export MRRC_PRODUCT_NAME=$productName" >> $mrrcEnvFile
+        echo "export MRRC_PRODUCT_VERSION=$productVersion" >> $mrrcEnvFile
+
+        awsSecret=$(yq -r '.mrrc.awsSecret' $(params.dataJsonPath))
+        echo -n "$awsSecret" > "$(results.charonAWSSecret.path)"
+
+        consignPubKey=$(yq -r '.cosignPubKeyFile' $(params.dataJsonPath))
```

Comment:
We already have `.sign.cosignSecretName` elsewhere - for cosign signing of images. I assume this will be a different secret. So maybe better to have it under `.mrrc` to make it clearer what it's related to?

Also, what is the key file name for? Is that something that can be different every time? Isn't it always the same?

Reply:
Thanks for the explanation. Yeah, it makes sense. Since only public keys will be involved here, it can probably be a configMap (doesn't have to be secret). But I guess it doesn't make much difference - either way it will be mounted as a volume.

Regarding the key name, I wonder if it would make sense to stick to just one name (like cosign.pub) - there is probably no benefit in having a custom name, right? I mean it shouldn't even matter which file the key value is saved to - as long as the file is passed as a parameter to whatever needs the key file.

Reply:
> Thanks for the explanation. Yeah, it makes sense. Since only public keys will be involved here, it can probably be a configMap (doesn't have to be secret). But I guess it doesn't make much difference - either way it will be mounted as a volume.

Yes the configmap is ok. This will be used later in the following tasks

> Regarding the key name, I wonder if it would make sense to stick to just one name (like cosign.pub) - there is probably no benefit in having a custom name, right? I mean it shouldn't even matter which file the key value is saved to - as long as the file is passed as a parameter to whatever needs the key file.

Sounds good. Will remove this result and use a default file name in following tasks.

Reply:
Fixed in new commit